lib/helpers/populate/assignRawDocsToIdStructure.js

```diff
@@ -55,7 +55,7 @@ function assignRawDocsToIdStructure(rawIds, resultDocs, resultOrder, options, re
     doc = resultDocs[sid];
     // If user wants separate copies of same doc, use this option
     if (options.clone) {
-      doc = doc.constructor.hydrate(doc._doc);
+      doc = Model.hydrate(options.lean ? doc : doc._doc);
```

Comment:
So this fixes the error, but this also makes it so that the child doc gets hydrated despite `lean()`. If memory serves, `lean()` propagates to `populate()` queries as well, so the expected behavior is that we don't hydrate the populated doc. We would need to clone it using `utils.clone()`.

Reply:
Ah, good catch!

I changed that in the last commit, but for some reason using `doc = utils.clone(doc)` makes it so that the document is not assigned to the array. I verified that it does return a different copy from `doc`.

Can you think of a reason why it would do that?

Reply:
No I'm not sure, I would need to do some debugging. I'll take a look later this week if you can't figure it out.

Reply:
Yep, I spent 3 good hours trying to figure it out with no luck. Will need your help with this one.